ml-algorithms/src/main/java/org/opensearch/ml/engine/encryptor/EncryptorImpl.java

```diff
@@ -351,7 +363,8 @@ private void handlePutDataObjectFailure(
         CountDownLatch latch
     ) {
         Exception cause = SdkClientUtils.unwrapAndConvertToException(throwable, OpenSearchStatusException.class);
-        if (cause instanceof VersionConflictEngineException) {
+        if (cause instanceof VersionConflictEngineException
```

Comment:
For DDB client this conditional works because there is no wrapped exception, but for Remote Client, using `OpenSearchStatusException` on line 365 will result in `cause` being an `OpenSearchException` if that is left as-is.

Two options to fix:
1. Just remove that `OpenSearchStatusException.class` vararg on line 365
2. Change this conditional to be `instanceof OpenSearchException` (since OpenSearchStatusException extends it, this works).